Archive activities in the portal  (for WaterScience)

knowuh opened the pull request:
This PR is still in progress.

scytacki commented on app/assets/javascripts/components/materials_bin/_fetch_data_mixin.js.coffee:
```diff
@@ -35,5 +35,25 @@ window.MBFetchDataMixin =
           newState[@dataStateKey] = if @processData? then @processData(data) else data
           @setState newState
 
+  archive: (material_id, arhive_url) ->
```
typo in `arhive_url`

scytacki commented on app/assets/javascripts/components/materials_bin/_fetch_data_mixin.js.coffee:
```diff
@@ -35,5 +35,25 @@ window.MBFetchDataMixin =
+          copy
+        @setState(collectionsData: newState)
+
+  archiveSingle: (material_id, arhive_url) ->
```
typo in `arhive_url`

scytacki commented on app/controllers/external_activities_controller.rb:
```diff
@@ -288,6 +288,20 @@ def matedit
+    authorize @external_activity
+    @external_activity.archive!
+    flash[:notice]= t("matedit.archive_success", {name: @external_activity.name})
+    redirect_to :search  #TBD:  Where to go?
```
I don't know if this was cleared up in the meeting. But I agree that going to search is the best place.

knowuh replied:
The one problem with this at the moment, is that the flash message doesn't seem to display on the search page.   I am looking into it.

knowuh commented on spec/models/portal/offering_spec.rb:
```diff
@@ -1,70 +1,69 @@
+       offering.student_report_enabled?.should be_true
       end
 
       it "is enabled" do
```
Just realized this "it" should be "is disabled"

scytacki commented on app/views/external_activities/matedit.html.haml:
```diff
@@ -6,8 +6,18 @@
+      = link_to t("matedit.archive"),
+        archive_external_activity_path(@external_activity),
+        confirm: t("matedit.archive_confirm"),
+        :class=>"button"
```
Did Jen and Carolyn like having this be a button? I was thinking a less prominent link would be better. I'm fine with a button if they liked it.

knowuh replied:
They liked it.  The didn't love the confirmation box, so I am working on a fix for that.

scytacki commented:
This looks good to me.

knowuh commented:
@scytacki I added a few more commits. 

As far as I can tell the failing cuke tests are the usual suspects:

>`cucumber features/teacher_manages_class.feature:54`
>`cucumber features/authors_can_filter_their_own_material.feature:21 `

**Changes in these commits are:**

1. new confirmation dialog box to replace system 'confirm'.
2. fixed appearance of (archive this) text.
3. hid archived activities from a few more views.
4. add a new icon to indicate archived materials in collections management view.

knowuh commented:
It turns out there is another story to deliver as part of this PR.  Please hold off on the review.

scytacki commented:
@knowuh This has a conflict now. Probably it is a conflict in schema.rb since I just merged the image library change which changed schema.rb.

knowuh commented:
@scytacki -- merged master in, keeping your `schema.rb` changes in manual merge.
